src/qs8-packw/gen/qs8-packw-x8c8-gemm-goi-wasmrelaxedsimd.c

```diff
@@ -0,0 +1,370 @@
+// Auto-generated file. Do not edit!
+//   Template: src/x8-packw/kr-wasmdot.c.in
+//   Generator: tools/xngen
+//
+// Copyright 2024 Google LLC
+//
+// This source code is licensed under the BSD-style license found in the
+// LICENSE file in the root directory of this source tree.
+
+
+#include <assert.h>
+
+#include <wasm_simd128.h>
+
+#include "xnnpack/packw.h"
+
+
+void xnn_qs8_packw_gemm_goi_ukernel_x8c8__wasmrelaxedsimd(
+  size_t g,
+  size_t nc,
+  size_t kc,
+  size_t nr,
+  size_t kr,
+  size_t sr,
+  const int8_t* weights,
+  const int32_t* bias,
+  const void* scale,
+  int8_t* packed_weights,
+  size_t extra_bytes,
+  const void* params)
+{
+  assert(g != 0);
+  assert(nc != 0);
+  assert(kc != 0);
+  assert(nr == 8);
+  assert(kr == 8);
+  assert(sr == 1);
+  assert(weights != NULL);
+  assert(packed_weights != NULL);
+
+  const v128_t vone = wasm_i8x16_splat(1);
+  const v128_t vzero = wasm_i32x4_splat(0);
+  XNN_FORCE_REALIZATION(vone);
+  XNN_FORCE_REALIZATION(vzero);
+  int8_t* out = (int8_t*) packed_weights;
+  const uint32_t* b = (const uint32_t*) bias;
+  const uint32_t izp = (uint32_t) (params ? (((const struct xnn_qs8_packw_params*) params)->input_zero_point + 0): 0);
+  v128_t vzeropoint = wasm_i32x4_splat((int32_t) izp);
+
+  do {
+    // NC main loop multiple of 8
+    const int8_t* w0 = (const int8_t*) weights;
+    size_t n = nc;
+    for (;n >= 8; n -= 8) {
+      int32_t* packed_b = (int32_t*) out;
+      if XNN_LIKELY(b != NULL) {
+        const v128_t vb0 = wasm_v128_load(b + 0);
+        wasm_v128_store(out + 0, vb0);
+        const v128_t vb1 = wasm_v128_load(b + 4);
+        wasm_v128_store(out + 16, vb1);
+        b += 8;
+      } else {
+        wasm_v128_store(out + 0, vzero);
+        wasm_v128_store(out + 16, vzero);
+      }
+      out += 8 * sizeof(uint32_t);
+
+      const int8_t* w1 = w0 + kc;
+      const int8_t* w2 = w1 + kc;
+      const int8_t* w3 = w2 + kc;
+      const int8_t* w4 = w3 + kc;
+      const int8_t* w5 = w4 + kc;
+      const int8_t* w6 = w5 + kc;
+      const int8_t* w7 = w6 + kc;
+
+      v128_t vacc01 = wasm_i32x4_splat(0);
+      v128_t vacc23 = wasm_i32x4_splat(0);
+      v128_t vacc45 = wasm_i32x4_splat(0);
+      v128_t vacc67 = wasm_i32x4_splat(0);
+
+      // KC main loop multiple of 8x8
+      size_t k = kc;
+      for (; k >= 16; k -= 16) {
+        v128_t v0_01 = wasm_v128_load(w0);
+        v128_t v1_01 = wasm_v128_load(w1);
+        v128_t v2_01 = wasm_v128_load(w2);
+        v128_t v3_01 = wasm_v128_load(w3);
+        v128_t v4_01 = wasm_v128_load(w4);
+        v128_t v5_01 = wasm_v128_load(w5);
+        v128_t v6_01 = wasm_v128_load(w6);
+        v128_t v7_01 = wasm_v128_load(w7);
+
+        v128_t v01_0 = wasm_i64x2_shuffle(v0_01, v1_01, 0, 2);
```

Comment:
i expect revec can't handle this.
for avx I was thinking of doing 4 columns, but its complicated.  the first 2 columns are as you have it.  the next 2 columns i think can be done with insertf128